test/leafTest.js

```diff
@@ -0,0 +1,151 @@
+const { BigNumber } = require("ethers")
+const { expect } = require("chai")
+
+describe("Leaf", () => {
+  let leafInstance
+
+  beforeEach(async () => {
+    const LeafStub = await ethers.getContractFactory("LeafStub")
+    leafInstance = await LeafStub.deploy()
+    await leafInstance.deployed()
+  })
+
+  describe("make", async () => {
```

Comment:
Are non-happy paths in scope of this PR? If so, we can cover them here with the following structure:
```
describe("make")
    context("when creation block does not exceed the uint64 max value")
        context("when ID does not exceed the uint32 max value")
            it("constructs a leaf with proper structure (address, creationBlock, id)")
        context("when ID exceeds the uint32 max value")
            it("should revert")
    context("when creation block exceeds the uint64 max value")
        it("should revert")
```
or alternatively, using a more flat but less explanatory structure:
```
describe("make")
    context("when all parameters are valid")
        it("constructs a leaf with proper structure (address, creationBlock, id)")
    context("when ID exceeds the uint32 max value")
        it("should revert")
    context("when creation block exceeds the uint64 max value")
        it("should revert")
```

Reply:
out of scope for this particular PR, but this is super useful and I'll add a follow-up